tests/eko/io/test_legacy.py

```diff
@@ -1,20 +1,84 @@
 import pathlib
-
-TEST = pathlib.Path(__file__).parents[2] / "data"
+from math import nan
+TEST = pathlib.Path(__file__).parents[2] / "data"   # directory of the EKO object
 
 import pathlib
 import eko
 from banana import toy
+import yaml
+import numpy as np
+from eko.interpolation import make_grid
+from ekobox.evol_pdf import evolve_pdfs
+import lhapdf
 
 pdf = toy.mkPDF("",0)
 
 from ekobox.apply import apply_pdf
 
-#print(evolved_pdfs.keys())
+x_grid = [1E-7,1E-6,1E-5,1E-4,1E-3,1E-2,0.1,0.3,0.5,0.7,0.9]
+Q2 = 10000
+
 def test_read_legacy():
-    for name in ["v0.13.tar"]:
-        with eko.EKO.read(TEST/name) as evolution_operator:           # directory of the EKO object
-            evolved_pdfs, _integration_errors = apply_pdf(evolution_operator, pdf)
+    for name in ["v0.14.tar"]:
+    
+        with eko.EKO.read(TEST/name) as evolution_operator:           
+            #evolved_pdfs, _integration_errors = apply_pdf(evolution_operator, pdf)
             #import pdb; pdb.set_trace()
             #print('test')
-            assert isinstance(evolution_operator.theory_card, eko.io.runcards.TheoryCard)
\ No newline at end of file
+            assert isinstance(evolution_operator.theory_card, eko.io.runcards.TheoryCard)   # Check that theory card is read as theory card
+            assert isinstance(evolution_operator.operator_card, eko.io.runcards.OperatorCard)   # Check that operator card is read as operator card
+        
+            # Set up the operator card for the interpolation
+            op_card = evolution_operator.operator_card
+            op_card.xgrid = x_grid
+            #op_card.mugrid = [(float(q), 5) for q in np.geomspace(5.0, 100, 5)]
+            op_card.init = evolution_operator.metadata.origin
+            
+            # Set up the theory card
+            th_card = evolution_operator.theory_card
+            
+            print(th_card.raw)
+            print(op_card.raw)
+
+        path = pathlib.Path(TEST/name)
+        path.unlink(missing_ok=True)
+        eko.solve(th_card,op_card,path)
+        evolve_pdfs(
+            [pdf], th_card, op_card, install=True, name="Evolved_pdf", store_path=path
+        )
+    
+        evolved_pdf = lhapdf.mkPDF("Evolved_pdf_{name}", 0)
+        
+        
+        #pdf_test = evolved_pdfs[10000,4][21]    # evolved gluon PDF at 10000 GeV^2
+        pdf_test = []
+        for k in range(len(x_grid)):
+            pdf_test.append(evolved_pdf.xfxQ2(21, x_grid[k], Q2))
+
+        # Import the values of the LHA benchmark tables. This is not very nice yet, but should work
+        lha_path = pathlib.Path(__file__).parents[4] / 'eko/src/ekomark/benchmark/external/LHA.yaml'
+        with open(lha_path,'r') as file:
+            lha_benchmark = yaml.safe_load(file)
+
+        xpdf_benchmark = lha_benchmark["table2"]["part2"]["g"]        
```

Comment:
use the `ekomark` interface to do all this - it knows and you don't need to hack here